C/Coin-OR/Osi/build_tarballs.jl

```diff
@@ -74,5 +83,5 @@ build_tarballs(
     products,
     dependencies;
     preferred_gcc_version = gcc_version,
-    julia_compat = "1.6",
+    julia_compat = "1.9",
```

Comment:
Do we need a preferred llvm version here so that it matches the CoinUtils one?

Reply:
I think that we need is `clang_use_lld = false`.
The issue is the new linker on Mac that doesn't support some options.
Mosè added this option after my recipe for `CoinUtils`.

But using an older version of `llvm` should also ensure that.